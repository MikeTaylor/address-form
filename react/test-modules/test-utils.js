import React from 'react'
import * as reactTestingLibrary from 'react-testing-library'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import { IntlProvider, intlShape } from 'react-intl'
import defaultStrings from '../../messages/en.json'
import enCountryCodeTranslations from 'i18n-iso-countries/langs/en.json'
import reduce from 'lodash/reduce'
import { getISOAlpha3 } from '../../demo/src/countryISO'

function addCountryCodeNameSpace(obj) {
  return reduce(
    obj,
    (acc, value, key) => {
      acc[`country.${getISOAlpha3(key)}`] = value
      return acc
    },
    {}
  )
}

const messages = {
  ...defaultStrings,
  ...addCountryCodeNameSpace(enCountryCodeTranslations),
}

const customRender = (node, options) => {
  const rendered = reactTestingLibrary.render(
    <IntlProvider
      messages={messages}
      locale="en-US">
      {node}
    </IntlProvider>,
    options
  )

  return {
    ...rendered,
    rerender: newUi =>
      customRender(newUi, {
        container: rendered.container,
        baseElement: rendered.baseElement,
      }),
  }
}

function customMount(node, { context, childContextTypes } = {}) {
  const intlProvider = new IntlProvider({
    locale: 'en-US',
    messages,
  }, {})

  const { intl } = intlProvider.getChildContext()

  return mount(
    React.cloneElement(node, { intl }),
    {
      context: Object.assign({}, context, {intl}),
      childContextTypes: Object.assign({}, { intl: intlShape }, childContextTypes),
    }
  )
}

function customShallow(node, options = { context: {}}) {
  const intlProvider = new IntlProvider({
    locale: 'en-US',
    messages,
  }, {})

  const { intl } = intlProvider.getChildContext()

  return shallow(
    React.cloneElement(node, { intl }),
    { ...options, context: { ...options.context, intl } }
  )
}

// re-export everything
module.exports = {
  ...reactTestingLibrary,
  render: customRender,
  mount: customMount,
  shallow: customShallow,
  rendererCreate(node) {
    return renderer.create(
      <IntlProvider
        messages={messages}
        locale="en-US">
        {node}
      </IntlProvider>
    )
  },
}
