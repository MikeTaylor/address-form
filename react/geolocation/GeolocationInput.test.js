
import React from 'react'
import { rendererCreate } from 'test-utils'

import GeolocationInput from './GeolocationInput'
import mockAddress from '../__mocks__/newAddress'

const DEFAULT_PROPS = {
  address: mockAddress,
  loadingGoogle: true,
  onChangeAddress: jest.fn(),
  rules:{
    country: 'BRA',
  },
}

describe('GeolocationInput', () => {
  it('should render default case', () => {
    const tree = rendererCreate(<GeolocationInput {...DEFAULT_PROPS} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render with changed label', () => {
    const props = {
      ...DEFAULT_PROPS,
      inputProps: {
        field: {
          fixedLabel: 'Teste',
        },
      },
    }
    const tree = rendererCreate(<GeolocationInput {...props} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
