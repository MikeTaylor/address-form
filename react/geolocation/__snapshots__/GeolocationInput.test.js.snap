// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeolocationInput should render with changed label 1`] = `
<p
  className="input ship-addressQuery text"
>
  <label
    htmlFor="ship-addressQuery"
  >
    Teste
  </label>
  <input
    autoComplete="on"
    autoFocus={false}
    className="loading-postal-code"
    data-hj-whitelist={true}
    disabled={true}
    id="ship-addressQuery"
    maxLength={undefined}
    name="addressQuery"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Eg: 225 East 41st Street, New York"
    type="text"
    value=""
  />
  <i
    className="loading-inline icon-spinner icon-spin"
  >
    <span>
      Loading
    </span>
  </i>
</p>
`;

exports[`GeolocationInput should render with default label 1`] = `
<p
  className="input ship-addressQuery text"
>
  <label
    htmlFor="ship-addressQuery"
  >
    Address
  </label>
  <input
    autoComplete="on"
    autoFocus={false}
    className="loading-postal-code"
    data-hj-whitelist={true}
    disabled={true}
    id="ship-addressQuery"
    maxLength={undefined}
    name="addressQuery"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Eg: 225 East 41st Street, New York"
    type="text"
    value=""
  />
  <i
    className="loading-inline icon-spinner icon-spin"
  >
    <span>
      Loading
    </span>
  </i>
</p>
`;
